geopandas/tests/test_geocode.py

```diff
@@ -58,94 +55,100 @@ def __call__(self, *args, **kwargs):
-                assert len(self.locations) == m.call_count
-
-            n = len(self.locations)
-            assert isinstance(g, GeoDataFrame)
-            expected = GeoSeries(
-                [Point(float(x) + 0.5, float(x)) for x in range(n)],
-                crs=from_epsg(4326))
-            assert_geoseries_equal(expected, g['geometry'])
-            assert_series_equal(g['address'],
-                                pd.Series(self.locations, name='address'))
-
-    def test_reverse(self):
-        from geopy.geocoders import GoogleV3
-        for provider in ['googlev3', GoogleV3]:
-            with mock.patch('geopy.geocoders.googlev3.GoogleV3.reverse',
-                            ReverseMock()) as m:
-                g = reverse_geocode(self.points, provider=provider, timeout=2)
-                assert len(self.points) == m.call_count
-
-            assert isinstance(g, GeoDataFrame)
-
-            expected = GeoSeries(self.points, crs=from_epsg(4326))
-            assert_geoseries_equal(expected, g['geometry'])
-            address = pd.Series(
-                ['address' + str(x) for x in range(len(self.points))],
-                name='address')
-            assert_series_equal(g['address'], address)
+@pytest.fixture
+def locations():
+    locations = ['260 Broadway, New York, NY',
+                 '77 Massachusetts Ave, Cambridge, MA']
+    return locations
+
+
+@pytest.fixture
+def points():
+    points = [Point(-71.0597732, 42.3584308),
+              Point(-77.0365305, 38.8977332)]
+    return points
+
+
+def test_prepare_result():
+    # Calls _prepare_result with sample results from the geocoder call
+    # loop
+    p0 = Point(12.3, -45.6)  # Treat these as lat/lon
+    p1 = Point(-23.4, 56.7)
+    d = {'a': ('address0', p0.coords[0]),
+         'b': ('address1', p1.coords[0])}
+
+    df = _prepare_geocode_result(d)
+    assert type(df) is GeoDataFrame
+    assert from_epsg(4326) == df.crs
+    assert len(df) == 2
+    assert 'address' in df
+
+    coords = df.loc['a']['geometry'].coords[0]
+    test = p0.coords[0]
+    # Output from the df should be lon/lat
+    assert coords[0] == pytest.approx(test[1])
+    assert coords[1] == pytest.approx(test[0])
+
+    coords = df.loc['b']['geometry'].coords[0]
+    test = p1.coords[0]
+    assert coords[0] == pytest.approx(test[1])
+    assert coords[1] == pytest.approx(test[0])
+
+def test_prepare_result_none():
+    p0 = Point(12.3, -45.6)  # Treat these as lat/lon
+    d = {'a': ('address0', p0.coords[0]),
+         'b': (None, None)}
+
+    df = _prepare_geocode_result(d)
+    assert type(df) is GeoDataFrame
+    assert from_epsg(4326) == df.crs
+    assert len(df) == 2
+    assert 'address' in df
+
+    row = df.loc['b']
+    assert len(row['geometry'].coords) == 0
+    assert np.isnan(row['address'])
+
+def test_bad_provider_forward():
+    from geopy.exc import GeocoderNotFound
+     with pytest.raises(GeocoderNotFound):
```

Comment:
you have a wrong identation here (this line is one space off)

Reply:
oops! fixed.